From 2497803bf05855ffb703b72758da1250dedd0ab4 Mon Sep 17 00:00:00 2001
From: mujiburrahimank <mujiburrahimank@maxlinear.com>
Date: Wed, 20 Nov 2024 18:39:16 +0800
Subject: [PATCH] WLANRTSYS-85697 [OCT][Func][UPDK][MLO][MFR]: After MLSR
 connection is established b/w two radio's, MFR value of opposite radio gets
 changed to default rate(for example:6mbps for 5g/6g & 1 mbps for 2g once the
 other radio's MFR is changed manually

Issue: MFR value of sibling VAP restores to default when the main VAP in MLO goes for toggle.

RCA: When MFR (Management Frames Rate) is set for any MLO VAP the corresponding sibling VAP will be toggled which sets the default rates of MFR.

Fix: While setting up MLO links explicity set the MFR rates for sibling VAP.

Fix up to WLANRTSYS-60843: MLO changes: AP MLD configuration

Signed-Off-By : Mujibur Rahiman K <mujiburrahimank@maxlinear.com>
---
 src/ap/mxl_mld.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/ap/mxl_mld.c b/src/ap/mxl_mld.c
index 9b4d054cf..3fe8b102a 100644
--- a/src/ap/mxl_mld.c
+++ b/src/ap/mxl_mld.c
@@ -305,6 +305,12 @@ static int mxl_setup_link(struct p_ap_mld *mld)
 			}
 		}
 	}
+	/* Set Management Frames Rate for MLD VAPs*/
+	if (hostapd_drv_set_management_frames_rate(link1, (u8)link1->conf->management_frames_rate) < 0)
+		wpa_printf(MSG_ERROR, "MXL setup link 1 failed to set Management Frames Rate ");
+
+	if (hostapd_drv_set_management_frames_rate(link2, (u8)link2->conf->management_frames_rate) < 0)
+		wpa_printf(MSG_ERROR, "MXL setup link 2 failed to set Management Frames Rate ");
 
 	ret = ieee802_11_set_beacon(link1);
 	if (ret < 0)
-- 
2.43.0

